frappe/public/js/frappe/model/create_new.js

```diff
@@ -283,23 +283,24 @@ $.extend(frappe.model, {
 	},
 
 	copy_doc: function (doc, from_amend, parent_doc, parentfield) {
-		var no_copy_list = ["name", "amended_from", "amendment_date", "cancel_reason"];
-		var newdoc = frappe.model.get_new_doc(doc.doctype, parent_doc, parentfield);
+		let no_copy_list = ["name", "amended_from", "amendment_date", "cancel_reason"];
+		let newdoc = frappe.model.get_new_doc(doc.doctype, parent_doc, parentfield);
 
-		for (var key in doc) {
-			// dont copy name and blank fields
-			var df = frappe.meta.get_docfield(doc.doctype, key);
+		for (let key in doc) {
+			// don't copy name and blank fields
+			let df = frappe.meta.get_docfield(doc.doctype, key);
 
 			if (
 				df &&
-				key.substr(0, 2) != "__" &&
+				key.substring(0, 2) != "__" &&
 				!no_copy_list.includes(key) &&
-				!(df && !from_amend && cint(df.no_copy) == 1)
+				!(df && !from_amend && cint(df.no_copy) == 1) &&
```

Comment:
This is hard to read and understand right now, let me refactor this particular conditional.